src/services/Viewer.js

```diff
@@ -157,13 +168,14 @@ export default class Viewer {
 	 * @param {object} options Options for opening the viewer
 	 * @param {string} options.path path of the file to open
 	 * @param {object[]} [options.list] the list of files as objects (fileinfo) format
+	 * @param {string} options.el selector of the element to render the file in
 	 * @param {Function} options.loadMore callback for loading more files
 	 * @param {boolean} options.canLoop can the viewer loop over the array
 	 * @param {Function} options.onPrev callback when navigating back to previous file
 	 * @param {Function} options.onNext callback when navigation forward to next file
 	 * @param {Function} options.onClose callback when closing the viewer
 	 */
-	open({ path, list = [], loadMore = () => ([]), canLoop = true, onPrev = () => {}, onNext = () => {}, onClose = () => {} } = {}) {
+	open({ path, list = [], el = null, loadMore = () => ([]), canLoop = true, onPrev = () => {}, onNext = () => {}, onClose = () => {} } = {}) {
```

Comment:
I would prefer a dedicated method to define the root Element of the Viewer
Something like `OCA.Viewer.setRootElement(el)` ?

Reply:
✅